libiop/protocols/encoded/lincheck/basic_lincheck_aux.tcc

```diff
@@ -92,9 +93,24 @@ std::shared_ptr<std::vector<FieldT>> multi_lincheck_virtual_oracle<FieldT>::eval
     /* p_{alpha}^1 in [BCRSVW18], but now using the lagrange polynomial from 
      * [TODO: cite Succinct Aurora] instead of powers of alpha. */
     /* Compute p_alpha_prime. */
-    std::vector<FieldT> p_alpha_prime_over_codeword_domain = 
+    std::vector<FieldT> p_alpha_prime_over_codeword_domain;
+    std::vector<FieldT> vd_vp_evaluations;
+    std::vector<FieldT> cd_vp_evaluations;
+    if (this->variable_domain_.num_elements() > this->constraint_domain_.num_elements()){
+        p_alpha_prime_over_codeword_domain = 
         this->p_alpha_.evaluations_over_field_subset(this->codeword_domain_);
 
+        vd_vp_evaluations = this->vd_vp_.evaluations_over_field_subset(this->codeword_domain_);
+        cd_vp_evaluations = this->cd_vp_.evaluations_over_field_subset(this->codeword_domain_);
+
+        for (int i = 0; i < vd_vp_evaluations.size(); i++)
+            p_alpha_prime_over_codeword_domain[i] *= vd_vp_evaluations[i] * cd_vp_evaluations[i].inverse();
```

Comment:
This is actually going to be super slow on each element! (Inversions are relatively high execution time)

Can you batch inverse the `cd_vp_evaluations`? Here is an example of how you'd do that: https://github.com/scipr-lab/libiop/blob/cb63d0c292192af18caa77bf35eb3c534318f088/libiop/protocols/encoded/common/rowcheck.tcc#L36